src/Symfony/Component/String/Tests/AbstractAsciiTestCase.php

```diff
@@ -0,0 +1,730 @@
+<?php
+
+namespace Symfony\Component\String\Tests;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\String\AbstractString;
+use Symfony\Component\String\Exception\InvalidArgumentException;
+
+abstract class AbstractAsciiTestCase extends TestCase
+{
+    protected static function createFromString(string $string): AbstractString
+    {
+        throw new \BadMethodCallException('This method must be implemented by subclasses.');
+    }
+
+    public function testCreateFromString()
+    {
+        $bytes = static::createFromString('Symfony is a PHP framework!');
+
+        $this->assertSame('Symfony is a PHP framework!', (string) $bytes);
```

Comment:
Why use `$this` ? It's a `static` function

https://github.com/sebastianbergmann/phpunit/blob/master/src/Framework/Assert.php#L1579-L1586

Reply:
because that's consistent with the existing code base